tests/test-doc-portal.c

```diff
@@ -826,7 +826,7 @@ rm_rf_dir (GFile         *dir,
 
   if (temp_error != NULL)
     {
-      g_propagate_error (error, temp_error);
+      g_propagate_error (error, g_steal_pointer (&temp_error));
```

Comment:
Could turn temp_error into a autoptr while at it here and in the other cases as well.

Reply:
Yes please, but as a separate PR please. Let's get the bugs fixed first.

(I can't help wondering how much any of these code paths have been tested, even manually...)